crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -60,7 +59,9 @@ impl<'tcx> CodegenCx<'tcx> {
             SpirvType::Integer(32, true) => self
                 .builder
                 .def_constant(SpirvConst::U32(ty, val as i64 as i32 as u32)),
-            SpirvType::Integer(64, true) => self.builder.def_constant(SpirvConst::U64(ty, val)),
+            SpirvType::Integer(64, false) | SpirvType::Integer(64, true) => {
```

Comment:
This can be collapsed further into `SpirvType::Integer(64, _)` - note that the `32`-bit case can also be collapsed, if you replace `val as i64 as i32 as u32` with `val as u32` (they're equivalent, no order of truncations is different from any other, since you're chopping off bits - {sign,zero}-extension can only happen when you're widening the type).

Reply:
sounds like a good idea